aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -149,7 +159,11 @@ void TensorIterator::compute_types() {
           op.dtype = op.tensor.scalar_type();
         } else {
           op.device = common_device;
-          op.dtype = common_dtype;
+          if (compute_common_inputs_dtype_ && op.is_output) {
+            op.dtype = op.tensor.scalar_type();
```

Comment:
So we've computed the common inputs dtype, but... we're not saving it anywhere?

Reply:
So after https://github.com/pytorch/pytorch/pull/22273 commom_dtype will be promoted to other inputs and outputs